3-extra/1-count-words.js

```diff
@@ -24,11 +24,11 @@
 */
 
 function countWords(string) {
-  const wordCount = {};
-
-  // write code here
-
-  return wordCount;
+  return (string ? string.split(" ") : []).reduce(
```

Comment:
This is a decent implementation and you used quite a few tricks here. However, I ague that you need to look for simpler (more readable) solutions:

```javascript
function countWords(string) {
  const wordCount = {};
  const words = string ? string.split(" ") : []
  for (const word of words) {
    wordCount[word] = 1 + (wordCount[word] || 0)
  }
  return wordCount
}
```